components/server/src/workspace/gitpod-server-impl.ts

```diff
@@ -2771,9 +2771,9 @@ export class GitpodServerImpl implements GitpodServerWithTracing, Disposable {
     ): Promise<AdminGetListResult<WorkspaceAndInstance>> {
         traceAPIParams(ctx, { req });
 
-        await this.guardAdminAccess("adminGetWorkspaces", { req }, Permission.ADMIN_WORKSPACES);
+        const admin = await this.guardAdminAccess("adminGetWorkspaces", { req }, Permission.ADMIN_WORKSPACES);
 
-        return await this.workspaceDb
+        const wss = await this.workspaceDb
```

Comment:
could we not use the `getWorkspaces` function you have introduced in [#18565](https://github.com/gitpod-io/gitpod/pull/18565)

Reply:
I did not consider that work the effort, as the shapes are totally different, and this would involve changes to the frontend.
I'd much rather start fresh on the frontend, throw theses methods out, and start from "scratch". But that is definitely way out of scope here.